crates/conformance-test-utils/src/madara.rs

```diff
@@ -0,0 +1,163 @@
+use reth_primitives::Bytes;
+use starknet::core::types::FieldElement;
+use starknet::core::utils::get_storage_var_address;
+
+use crate::types::{ContractAddress, Felt, StorageKey, StorageValue};
+
+pub fn genesis_load_bytecode(
+    bytecode: Bytes,
+    address: FieldElement,
+) -> Vec<((ContractAddress, StorageKey), StorageValue)> {
+    bytecode
+        .chunks(16)
+        .enumerate()
+        .map(|(i, x)| {
+            let mut storage_value = [0u8; 32];
+            storage_value[16..16 + x.len()].copy_from_slice(x);
+            let storage_value = FieldElement::from_bytes_be(&storage_value).unwrap().into(); //safe unwrap since x.len() == 16
+
+            let storage_key: Felt = get_storage_var_address("bytecode_", &[FieldElement::from(i)]).unwrap().into(); // safe unwrap since bytecode_ is all ascii
+
+            ((address.into(), storage_key), storage_value)
+        })
+        .collect()
+}
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use kakarot_rpc_core::contracts::contract_account::ContractAccount;
+    use kakarot_rpc_core::mock::constants::ACCOUNT_ADDRESS;
+    use kakarot_rpc_core::test_utils::constants::EOA_WALLET;
+    use kakarot_rpc_core::test_utils::deploy_helpers::{construct_kakarot_test_sequencer, deploy_kakarot_system};
+    use starknet::core::types::{BlockId as StarknetBlockId, BlockTag};
+    use starknet::providers::jsonrpc::HttpTransport as StarknetHttpTransport;
+    use starknet::providers::JsonRpcClient;
+    use starknet_api::core::ContractAddress as StarknetContractAddress;
+    use starknet_api::hash::StarkFelt;
+    use starknet_api::state::StorageKey;
+
+    const TEST_BYTECODE: &str = "0xf234567890abcdef1234567890abcdeff234567890abcdef1234567890abcdef";
```

Comment:
why do you start with `f` and not `1` ?

Reply:
can we define the constants in the tests they belong to?

Reply:
using the same bytes16 two times is a bit confusing and doesn't test well the fact that the chunks need to be splitted properly. plz use something like 0x111..111222...222 or any other example that will make use sure that slot0 has the first bytes16 and slot1 the second